CMakeLists.txt

```diff
@@ -56,85 +57,90 @@ if(RET AND NOT RET EQUAL 0)
 endif()
 
 add_library(tibiarc
-  "lib/canvas.c"
-  "lib/canvas.h"
-  "lib/characterset.h"
-  "lib/container.c"
-  "lib/container.h"
-  "lib/creature.c"
-  "lib/creature.h"
-  "lib/datareader.c"
-  "lib/datareader.h"
-  "lib/demuxer.c"
+  "lib/canvas.cpp"
+  "lib/canvas.hpp"
+  "lib/characterset.cpp"
+  "lib/characterset.hpp"
+  "lib/container.hpp"
+  "lib/creature.hpp"
+  "lib/datareader.hpp"
+  "lib/demuxer.hpp"
   "lib/deps/7z/7zTypes.h"
   "lib/deps/7z/Alloc.c"
   "lib/deps/7z/Alloc.h"
   "lib/deps/7z/Compiler.h"
   "lib/deps/7z/LzmaDec.c"
   "lib/deps/7z/LzmaDec.h"
   "lib/deps/7z/Precomp.h"
-  "lib/deps/uthash.h"
-  "lib/effect.h"
-  "lib/fonts.c"
-  "lib/fonts.h"
-  "lib/formats/cam.c"
-  "lib/formats/rec.c"
-  "lib/formats/tibiacast.c"
-  "lib/formats/tibiareplay.c"
-  "lib/formats/tmv2.c"
-  "lib/formats/tmv1.c"
-  "lib/formats/ttm.c"
-  "lib/formats/yatc.c"
-  "lib/gamestate.c"
-  "lib/gamestate.h"
-  "lib/icons.c"
-  "lib/icons.h"
-  "lib/map.h"
-  "lib/message.c"
-  "lib/message.h"
-  "lib/missile.h"
-  "lib/object.h"
-  "lib/packet.c"
-  "lib/parser.c"
-  "lib/parser.h"
-  "lib/pictures.c"
-  "lib/pictures.h"
-  "lib/pixel.h"
-  "lib/player.h"
-  "lib/position.h"
-  "lib/recordings.c"
-  "lib/recordings.h"
-  "lib/renderer.c"
-  "lib/renderer.h"
-  "lib/report.c"
-  "lib/report.h"
-  "lib/sprites.c"
-  "lib/sprites.h"
-  "lib/textrenderer.c"
-  "lib/textrenderer.h"
-  "lib/tile.c"
-  "lib/tile.h"
-  "lib/types.c"
-  "lib/types.h"
-  "lib/utils.h"
-  "lib/versions_decl.h"
-  "lib/versions.c"
-  "lib/versions.h"
+  "lib/effect.hpp"
+  "lib/events.cpp"
+  "lib/events.hpp"
+  "lib/fonts.cpp"
+  "lib/fonts.hpp"
+  "lib/formats/cam.cpp"
+  "lib/formats/rec.cpp"
+  "lib/formats/tibiacast.cpp"
+  "lib/formats/tibiareplay.cpp"
+  "lib/formats/tmv2.cpp"
+  "lib/formats/tmv1.cpp"
+  "lib/formats/ttm.cpp"
+  "lib/formats/yatc.cpp"
+  "lib/gamestate.cpp"
+  "lib/gamestate.hpp"
+  "lib/icons.cpp"
+  "lib/icons.hpp"
+  "lib/map.hpp"
+  "lib/message.cpp"
+  "lib/message.hpp"
+  "lib/missile.hpp"
+  "lib/object.hpp"
+  "lib/parser.cpp"
+  "lib/parser.hpp"
+  "lib/pictures.cpp"
+  "lib/pictures.hpp"
+  "lib/pixel.hpp"
+  "lib/player.hpp"
+  "lib/position.hpp"
+  "lib/recordings.cpp"
+  "lib/recordings.hpp"
+  "lib/renderer.cpp"
+  "lib/renderer.hpp"
+  "lib/sprites.cpp"
+  "lib/sprites.hpp"
+  "lib/textrenderer.cpp"
+  "lib/textrenderer.hpp"
+  "lib/tile.cpp"
+  "lib/tile.hpp"
+  "lib/types.cpp"
+  "lib/types.hpp"
+  "lib/utils.hpp"
+  "lib/versions_decl.hpp"
+  "lib/versions.cpp"
+  "lib/versions.hpp"
 )
 target_include_directories(tibiarc PUBLIC "lib")
 target_link_libraries(tibiarc PRIVATE m)
 target_compile_definitions(tibiarc PUBLIC "LEVEL1_DCACHE_LINESIZE=${LEVEL1_DCACHE_LINESIZE}")
 
+if(MSVC)
+  target_compile_options(tibiarc PRIVATE /W4 /WX)
+else()
+  target_link_libraries(tibiarc PRIVATE stdc++_libbacktrace)
```

Comment:
I happened to upgrade to GCC14 yesterday, and it seems that in 14.0 (or 14.1?) you need to link against `stdc++exp` instead of `stdc++_libbacktrace`. See https://stackoverflow.com/a/72342952

Not sure if there is a better way of picking the correct library in Cmake than just checking `CMAKE_CXX_COMPILER_VERSION `.

While writing this I also tried to compile with clang (19), but that seems to fail for some other reason (LTO?): `/usr/bin/ld: libtibiarc.a: error adding symbols: file format not recognized` when linking `miner`. `player` and `converter` links OK.

Reply:
Nevermind, I just had to add `miner` to the set_target_properties INTERPROCEDURAL_OPTIMIZATION thing on line 315 for clang to work.